src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/XMLASTAnalyser.java

```diff
@@ -0,0 +1,245 @@
+            String key = reader.getNamespacePrefix(i);
+            key = key == null ? "xmlns" : "xmlns:" + key;
+
+            final String value = reader.getNamespaceURI(i);
+
+            addAttribute(key, value);
+
+            if (i < namespaceCount - 1 || attributeCount > 0) {
+                addToken(new Token(WHITESPACE, " "));
+            }
+        }
+
+        if (attributeCount > 0) {
+            for (int i = 0; i < attributeCount; i++) {
+                final String prefix = reader.getAttributePrefix(i);
+                final String key = (prefix == "" ? "" : prefix + ":")  + reader.getAttributeLocalName(i);
+                final String value = reader.getAttributeValue(i);
+
+                addAttribute(key, value);
+
+                if (i < attributeCount - 1) {
+                    addToken(new Token(WHITESPACE, " "));
+                }
+            }
+
+            addToken(new Token(KEYWORD, ">"));
+        }
+    }
+
+    private void endElement(final XMLStreamReader reader) {
+        final String elementName = reader.getLocalName();
+        final String definitionId = elementName + "/" + reader.getLocation().getLineNumber();
+
+        if (lastToken != null && reader.getLocation().getCharacterOffset() == startElementCharOffset) {
+            // This element ends at the same place it starts, so it is a self-closing element, and we should try
+            // to retrofit in the self-closing '/>' into the previous token to more closely emulate the input file.
+            final String lastTokenValue = lastToken.getValue();
+            final String newValue = (lastTokenValue.endsWith(">")
+                 ? lastTokenValue.substring(0, lastTokenValue.length() - 1)
+                 : lastTokenValue)
+                 + " />";
+            lastToken.setValue(newValue);
+        } else {
+            addToken(new Token(KEYWORD, "</" + elementName + ">", definitionId));
+        }
+    }
+
+    private void addAttribute(String key, String value) {
+        addToken(new Token(TYPE_NAME, key));
+        addToken(new Token(STRING_LITERAL, "=\"" + value + "\""));
+    }
+
+    private void addCharacters(XMLStreamReader reader) {
+        final String characters = reader.getText();
+
+        int head = 0;
+        int tail = 0;
+        for (int i = 0; i < characters.length(); i++) {
+            char ch = characters.charAt(i);
+            if (ch == '\n') {
+                if (tail > head) {
+                    final String substring = characters.substring(head, tail);
+                    if (substring.trim().isEmpty()) {
+                        addToken(new Token(WHITESPACE, substring));
+                    } else {
+                        addToken(new Token(TEXT, substring));
+                    }
+                }
+                addNewLine();
+                head = i + 1;
+                tail = i + 1;
+            } else {
+                tail++;
+            }
```

Comment:
Can we do the following here or is there some special case that this is handling?

```java
Arrays.stream(characters.split("\n"))
        .forEach(token -> {
               if (token.trim().isEmpty()) {
                   addToken(new Token(WHITESPACE, token));
               } else {
                   addToken(new Token(TEXT, token));
               }
               addNewLine(); 
       });
```

Reply:
This results in incorrect output, so I'm inclined to leave the code as-is here.